api/v1alpha1/vaultauth_types.go

```diff
@@ -51,6 +51,15 @@ type VaultAuthConfigJWT struct {
 	TokenExpirationSeconds int64 `json:"tokenExpirationSeconds,omitempty"`
 }
 
+// VaultAuthConfigAppRole provides VaultAuth configuration options needed for authenticating to Vault.
+type VaultAuthConfigAppRole struct {
+	// RoleID (role_id) to use for authenticating to Vault.
+	RoleID string `json:"roleid"`
```

Comment:
nit: I think the json flag should be mixed case: `roleId` , like `serviceAccount` above.